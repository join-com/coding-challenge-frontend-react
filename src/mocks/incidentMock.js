const incidentMock = {
  id: 24134,
  title: "Stolen di Florino (red and white)",
  description: "",
  address: "Berlin, 10245, DE",
  occurred_at: 1402984800,
  updated_at: 1545551568,
  url: "https://bikewise.org/api/v1/incidents/24134",
  source: {
    name: "BikeIndex.org",
    html_url: "https://bikeindex.org/bikes/27159",
    api_url: "https://bikeindex.org/api/v1/bikes/27159"
  },
  media: {
    image_url: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/large_dsc_1293.jpg",
    image_url_thumb: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/small_dsc_1293.jpg"
  },
  location_type: null,
  location_description: null,
  type: "Theft",
  type_properties: null
};

export default incidentMock;
