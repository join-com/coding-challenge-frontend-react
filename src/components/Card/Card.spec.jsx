import React from "react";
import { shallow } from "enzyme";
import Component from "./index";
import incidentMock from "../../mocks/incidentMock";

describe("Button component test", () => {
  const defaultProps = {
    item: incidentMock
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return shallow(<Component {...props} />);
  };

  it("should match snapshot", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot without description and image", () => {
    const wrapper = getWrapper({
      item: {
        ...defaultProps.item,
        description: null,
        media: {}
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
