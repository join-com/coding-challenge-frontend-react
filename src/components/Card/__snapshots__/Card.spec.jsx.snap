// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test should match snapshot 1`] = `
<styled.div
  key="24134"
>
  <styled.img
    alt="Stolen di Florino (red and white)"
    src="https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/small_dsc_1293.jpg"
  />
  <styled.div>
    <Link
      replace={false}
      to="/case/24134"
    >
      <Heading
        bold={false}
        level={3}
        uppercase={false}
      >
        Stolen di Florino (red and white)
      </Heading>
    </Link>
    <Text
      bold={false}
      size="normal"
      uppercase={false}
    >
      Occurred at: 
      Di., 17. Juni 2014
       - 
      Berlin, 10245, DE
    </Text>
    <Text
      bold={false}
      size="normal"
      uppercase={false}
    >
      Updated at: 
      So., 23. Dez. 2018
    </Text>
  </styled.div>
</styled.div>
`;

exports[`Button component test should match snapshot without description and image 1`] = `
<styled.div
  key="24134"
>
  <styled.img
    alt="Stolen di Florino (red and white)"
    src="/bicycle.svg"
  />
  <styled.div>
    <Link
      replace={false}
      to="/case/24134"
    >
      <Heading
        bold={false}
        level={3}
        uppercase={false}
      >
        Stolen di Florino (red and white)
      </Heading>
    </Link>
    <Text
      bold={false}
      size="normal"
      uppercase={false}
    >
      Occurred at: 
      Di., 17. Juni 2014
       - 
      Berlin, 10245, DE
    </Text>
    <Text
      bold={false}
      size="normal"
      uppercase={false}
    >
      Updated at: 
      So., 23. Dez. 2018
    </Text>
  </styled.div>
</styled.div>
`;
